src/Container/Container.php

```diff
@@ -36,76 +41,122 @@ class Container implements ContainerInterface
      * @param mixed  $value
      * @return Container
      */
-    public function set($key, $value)
+    public function set(string $key, $value): self
     {
+        if (!(\is_string($value) || \is_object($value))) {
+            throw new \InvalidArgumentException('Value must be a class name, an object instance, or a callable');
+        }
```

Comment:
Why remove the ability to set any/scalar values?

Reply:
Mainly because of the conflict with interface-to-class mappings, in which case the value is a string. 

I also think configuration values and the like don't belong in the Container, but should be kept in separate objects, but that's just a personal opinion.

Reply:
I agree that config should be kept separately and I don't see any issues with removing support for scalar values here